Automattic/wp-calypso
# Subscription Management: Create a new route to access the Individual Subscription Page using blog_id

Closes https://github.com/Automattic/wp-calypso/issues/82895

## Proposed Changes

* Updated `useSiteSubscriptionDetailsQuery` hook to fetch subscription details from two different endpoints, depending on which param was passed (blog_id or subscription_id).
* Updated SiteSubscription component to accept `blogId` prop.
* Removed `useSiteSubscriptionQuery`, it was replaced by `useSiteSubscriptionDetailsQuery`.
* Created `/read/site/subscription/:blog_id` route to avoid collisions.

## Testing Instructions

* Apply this PR to your local
* Subscribe to a site and get the site ID
* Go to `http://calypso.localhost:3000/read/site/subscription/{site-id}`
* You should see the Individual Subscription page
* Click on "Manage all subscriptions" to go back to the subscriptions list
* Click on the same site to open the Individual Subscription page using the subscription ID
* You should see the same Individual Subscription page as before
* Make regression tests on the external portal (http://calypso.localhost:3000/subscriptions/sites)

## Pre-merge Checklist
